src/scriptdev2/scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp

```diff
@@ -598,8 +598,11 @@ struct boss_arugalAI : public ScriptedAI
         {
             if (!m_bAttacking)
             {
-                DoCastSpellIfCan(m_creature->getVictim(), SPELL_VOID_BOLT);
-                m_uiVoidboltTimer = urand(2900, 4800);
+                if (Unit* victim = m_creature->getVictim())
```

Comment:
But your comment about adding it to CreatureAI, would that then mean we could remove it from all the different scripts? If so, and not much heavier on the server then it may actually not be a bad idea, what do you think?